src/wp-includes/option.php

```diff
@@ -1133,6 +1142,94 @@ function delete_option( $option ) {
 	return false;
 }
 
+/**
+ *  Determines the appropriate autoload value for an option based on input.
+ *
+ *  This function checks the provided autoload value and returns a standardized
+ *  value ('yes', 'no', 'auto-yes', or 'auto-no') based on specific conditions.
+ *
+ * If no valid autoload value is provided, the function will check the size of the given option value
+ * and compare it to the maximum allowed size. If it ls larger, it will return 'auto-no', otherwise
+ * it will return 'auto-yes'.
+ *
+ * @since 6.5.0
+ * @access private
+ *
+ * @param string           $option   The name of the option.
+ * @param mixed            $value    The value of the option to check its autoload value.
+ * @param string|bool|null $autoload The autoload value to check.
+ *                                   Accepts 'yes'|true to enable or 'no'|false to disable, or
+ *                                   'auto-yes' and 'auto-no' for internal purposes.
+ *                                   Any other autoload value will be forced to either `auto-yes`
+ *                                   or `auto-no`.
```

Comment:
Per https://github.com/WordPress/wordpress-develop/pull/5671#issuecomment-1877564944, this should also support a plain `auto`, without any clear intent.
```suggestion
 *                                   'auto-yes', 'auto-no', or 'auto' for internal purposes.
 *                                   Any other autoload value will be forced to either `auto-yes`,
 *                                   `auto-no`, or `auto`.
```